api/src/pcapi/core/bookings/models.py

```diff
@@ -260,6 +260,16 @@ def userName(self) -> str:
     def email(self) -> str:
         return self.user.email
 
+    @property
+    def emailAtDateCreated(self) -> str:
+        email = self.user.email
+        for history in reversed(sorted(self.user.email_history, key=lambda history: history.creationDate)):
+            print(history.id, history.creationDate, self.id)
+            if self.dateCreated >= history.creationDate:
+                email = f"{history.newUserEmail}{history.newDomainEmail}"
+                break
```

Comment:
```suggestion
            else:
                break
```
A priori tu peux avoir plusieurs changements d'email depuis la date de réservation, donc plusieurs itérations de la boucle. Cela sera sûrement vérifié par les tests de cette propriété du modèle.

Reply:
Pas sur de comprendre la remarque, si je sorte par date ASC, le premier qui est après la date de création du booking est le bon email ?